app/src/main/kotlin/io/homeassistant/companion/android/settings/server/ServerSettingsFragment.kt

```diff
@@ -363,6 +387,25 @@ class ServerSettingsFragment :
 
         presenter.updateServerName()
         presenter.updateUrlStatus()
+        updateSecurityLevelSummary()
+        potentiallyShowSecurityLevel()
+    }
+
+    private fun potentiallyShowSecurityLevel() {
+        lifecycleScope.launch {
+            findPreference<Preference>("connection_security_level")?.let {
+                it.isVisible = USE_NEW_LAUNCHER && presenter.hasHTTPURL()
```

Comment:
This hides the setting even if I previously set it to allow insecure connections, and changed to an HTTPS URL. The setting is still set to allow insecure connections though, which I think is unwanted.

Reply:
@bgoncal didn't make this hide action. I think based on your comments it might make sense to also keep it. At least for now even on HTTPS